flash.sh

```diff
@@ -0,0 +1,298 @@
+	ADB=out/host/`uname -s | tr "[[:upper:]]" "[[:lower:]]"`-x86/bin/adb
+fi
+if [ ! -f "`which \"$FASTBOOT\"`" ]; then
+	FASTBOOT=out/host/`uname -s | tr "[[:upper:]]" "[[:lower:]]"`-x86/bin/fastboot
+fi
+
+run_adb()
+{
+	$ADB $ADB_FLAGS $@
+}
+
+update_time()
+{
+	if [ `uname` = Darwin ]; then
+		OFFSET=`date +%z`
+		OFFSET=${OFFSET:0:3}
+		TIMEZONE=`date +%Z$OFFSET|tr +- -+`
+	else
+		TIMEZONE=`date +%Z%:::z|tr +- -+`
+	fi
+	echo Attempting to set the time on the device
+	run_adb wait-for-device &&
+	run_adb shell "toolbox date $(date +%s) ||
+	               toolbox date -s $(date +%Y%m%d.%H%M%S)" &&
+	run_adb shell setprop persist.sys.timezone $TIMEZONE
+}
+
+sd_partition()
+{
+  DRIVE=$2
+
+  if [ -z "${DRIVE}" ]; then
+    echo You must specify a SD card drive
+  fi
+
+  umount $DRIVE
+
+  sudo dd if=/dev/zero of=$DRIVE bs=1024
+
+  sed -e 's/\t\([\+0-9a-zA-Z]*\)[ \t].*/\1/' << EOF | fdisk ${DRIVE}
+    o
+    n
+    p
+    1
+
+    +48M
+    n
+    p
+    2
+
+    +537M
+    n
+    p
+    3
+
+    +128M
+    n
+    p
+    4
+
+
+    a
+    1
+    p
+    w
+    q
+  EOF
+
+
+  case `uname` in
+  "Darwin")
+    e2fsck -f ${DRIVE}s4
+    ;;
+  "Linux")
+    e2fsck -f ${DRIVE}p4
+    ;;
+  *)
+    echo Unsupported platform: `uname`
+    exit -1
+  esac
+
+  case `uname` in
+  "Darwin")
+    resize2fs -f ${DRIVE}s4
+    ;;
+  "Linux")
+    resize2fs -f ${DRIVE}p4
+    ;;
+  *)
+    echo Unsupported platform: `uname`
+    exit -1
+  esac
+
+}
+
+write_images()
+{
+  DRIVE=$2
+
+  if [ -e "out/target/product/$DEVICE/boot.img" ]; then
+    case `uname` in
+    "Darwin")
+      dd if=out/target/product/${DEVICE}/boot.img of=${DRIVE}s1 bs=1m
+      ;;
+    "Linux")
+      dd if=out/target/product/${DEVICE}/boot.img of=${DRIVE}p1 bs=1M
+      ;;
+    *)
+      echo Unsupported platform: `uname`
+      exit -1
+    esac
+  fi
+
+  if [ -e "out/target/product/$DEVICE/system.img" ]; then
+    case `uname` in
+    "Darwin")
+      dd if=out/target/product/${DEVICE}/system.img of=${DRIVE}s2 bs=1m
+      ;;
+    "Linux")
+      dd if=out/target/product/${DEVICE}/system.img of=${DRIVE}p2 bs=1M
+      ;;
+    *)
+      echo Unsupported platform: `uname`
+      exit -1
+    esac
+  fi
+
+  if [ -e "out/target/product/$DEVICE/cache.img" ]; then
+    case `uname` in
+    "Darwin")
+      dd if=out/target/product/${DEVICE}/cache.img of=${DRIVE}s3 bs=1m
+      ;;
+    "Linux")
+      dd if=out/target/product/${DEVICE}/cache.img of=${DRIVE}p3 bs=1M
+      ;;
+    *)
+      echo Unsupported platform: `uname`
+      exit -1
+    esac
+  fi
+
+  if [ -e "out/target/product/$DEVICE/userdata.img" ]; then
+    case `uname` in
+    "Darwin")
+      dd if=out/target/product/${DEVICE}/userdata.img of=${DRIVE}s4 bs=1m
+      ;;
+    "Linux")
+      dd if=out/target/product/${DEVICE}/userdata.img of=${DRIVE}p4 bs=1M
+      ;;
+    *)
+      echo Unsupported platform: `uname`
+      exit -1
+    esac
+  fi
+}
+
+flash_image()
+{
+  DRIVE=$3
+
+  if [ -z "${DRIVE}" ]; then
```

Comment:
I didn't implemented, I talked with Atila and checking the device layout would be a plus after finishing script. Personally I really don't think that checking the device layout is a P1 for this script to be landed.